lib/phoenix/pubsub.ex

```diff
@@ -137,23 +142,23 @@ defmodule Phoenix.PubSub do
           PubSub.subscribe(MyApp.PubSub, "topic1",
             fastlane: {fast_pid, Phoenix.Transports.WebSocketSerializer, ["event1"]})
   """
-  @spec subscribe(atom, pid, binary) :: :ok | {:error, term}
-  def subscribe(server, pid, topic)
-    when is_atom(server) and is_pid(pid) and is_binary(topic) do
-    subscribe(server, pid, topic, [])
-  end
   @spec subscribe(atom, binary, Keyword.t) :: :ok | {:error, term}
   def subscribe(server, topic, opts)
     when is_atom(server) and is_binary(topic) and is_list(opts) do
     call(server, :subscribe, [self(), topic, opts])
   end
+  @spec subscribe(atom, pid, binary) :: :ok | {:error, term}
+  def subscribe(server, pid, topic)
+    when is_atom(server) and is_pid(pid) and is_binary(topic) do
+    subscribe(server, pid, topic, [])
+  end
   @spec subscribe(atom, binary) :: :ok | {:error, term}
```

Comment:
Because of how it rendered in [the docs](https://hexdocs.pm/phoenix_pubsub/Phoenix.PubSub.html#subscribe/3):

<img width="514" alt="screen shot 2017-05-16 at 7 15 33 am" src="https://cloud.githubusercontent.com/assets/49391/26110534/8222c84a-3a07-11e7-9d77-c3056b7960f2.png">

The docs are rendering `subscribe(server, pid, topic)` as the main function head for `subscribe/3`.  It's quite confusing because then you spell out 3 arguments which are different (`server`, `topic`, and `opts`).  At first I was quite confusing, then thought there was a typo in the docs when I saw this.  After looking at the code I realized that there are two `subscribe/3` function heads.  However, the one that takes a `pid` as the 2nd argument is deprecated since it delegates to `subscribe/4` which prints a deprecation warning.  So that explains why `pid` isn't included in the list of described arguments.  But it also means that we don't want the deprecated function head to be the one that ExDoc renders.  So I moved it down so that it'll now render `subscribe(server, topic, opts)` as the main function head, which aligns with the args you describe.

Of course, moving the function clause ordering bears some risk of changing the semantics of the code, but the two function heads I re-ordered have guard clauses for there are arguments and are mutually exclusive, so no call could ever match both.  Therefore, the ordering isn't being used for semantics here.